File: csharp/perfect-numbers/PerfectNumbersTests.cs
using System;
using Xunit;

public class PerfectNumbersTests
{
    [Fact]
    public void Smallest_perfect_number_is_classified_correctly()
    {
        Assert.Equal(Classification.Perfect, PerfectNumbers.Classify(6));
    }

    [Fact]
    public void Medium_perfect_number_is_classified_correctly()
    {
        Assert.Equal(Classification.Perfect, PerfectNumbers.Classify(28));
    }

    [Fact]
    public void Large_perfect_number_is_classified_correctly()
    {
        Assert.Equal(Classification.Perfect, PerfectNumbers.Classify(33550336));
    }

    [Fact]
    public void Smallest_abundant_number_is_classified_correctly()
    {
        Assert.Equal(Classification.Abundant, PerfectNumbers.Classify(12));
    }

    [Fact]
    public void Medium_abundant_number_is_classified_correctly()
    {
        Assert.Equal(Classification.Abundant, PerfectNumbers.Classify(30));
    }

    [Fact]
    public void Large_abundant_number_is_classified_correctly()
    {
        Assert.Equal(Classification.Abundant, PerfectNumbers.Classify(33550335));
    }

    [Fact]
    public void Smallest_prime_deficient_number_is_classified_correctly()
    {
        Assert.Equal(Classification.Deficient, PerfectNumbers.Classify(2));
    }

    [Fact]
    public void Smallest_non_prime_deficient_number_is_classified_correctly()
    {
        Assert.Equal(Classification.Deficient, PerfectNumbers.Classify(4));
    }

    [Fact]
    public void Medium_deficient_number_is_classified_correctly()
    {
        Assert.Equal(Classification.Deficient, PerfectNumbers.Classify(32));
    }

    [Fact]
    public void Large_deficient_number_is_classified_correctly()
    {
        Assert.Equal(Classification.Deficient, PerfectNumbers.Classify(33550337));
    }

    [Fact]
    public void Edge_case_no_factors_other_than_itself_is_classified_correctly()
    {
        Assert.Equal(Classification.Deficient, PerfectNumbers.Classify(1));
    }

    [Fact]
    public void Zero_is_rejected_as_it_is_not_a_positive_integer_()
    {
        Assert.Throws<ArgumentOutOfRangeException>(() => PerfectNumbers.Classify(0));
    }

    [Fact]
    public void Negative_integer_is_rejected_as_it_is_not_a_positive_integer_()
    {
        Assert.Throws<ArgumentOutOfRangeException>(() => PerfectNumbers.Classify(-1));
    }
}

File: vbnet/perfect-numbers/PerfectNumbersTests.vb
Imports System
Imports Xunit
Public Class PerfectNumbersTests
    <Fact>
    Public Sub SmallestPerfectNumberIsClassifiedCorrectly()
        Assert.Equal(Classification.Perfect, Classify(6))
    End Sub
    <Fact>
    Public Sub MediumPerfectNumberIsClassifiedCorrectly()
        Assert.Equal(Classification.Perfect, Classify(28))
    End Sub
    <Fact>
    Public Sub LargePerfectNumberIsClassifiedCorrectly()
        Assert.Equal(Classification.Perfect, Classify(33550336))
    End Sub
    <Fact>
    Public Sub SmallestAbundantNumberIsClassifiedCorrectly()
        Assert.Equal(Classification.Abundant, Classify(12))
    End Sub
    <Fact>
    Public Sub MediumAbundantNumberIsClassifiedCorrectly()
        Assert.Equal(Classification.Abundant, Classify(30))
    End Sub
    <Fact>
    Public Sub LargeAbundantNumberIsClassifiedCorrectly()
        Assert.Equal(Classification.Abundant, Classify(33550335))
    End Sub
    <Fact>
    Public Sub SmallestPrimeDeficientNumberIsClassifiedCorrectly()
        Assert.Equal(Classification.Deficient, Classify(2))
    End Sub
    <Fact>
    Public Sub SmallestNonPrimeDeficientNumberIsClassifiedCorrectly()
        Assert.Equal(Classification.Deficient, Classify(4))
    End Sub
    <Fact>
    Public Sub MediumDeficientNumberIsClassifiedCorrectly()
        Assert.Equal(Classification.Deficient, Classify(32))
    End Sub
    <Fact>
    Public Sub LargeDeficientNumberIsClassifiedCorrectly()
        Assert.Equal(Classification.Deficient, Classify(33550337))
    End Sub
    <Fact>
    Public Sub EdgeCaseNoFactorsOtherThanItselfIsClassifiedCorrectly()
        Assert.Equal(Classification.Deficient, Classify(1))
    End Sub
    <Fact>
    Public Sub ZeroIsRejectedNotANaturalNumber()
        Assert.Throws(Of ArgumentOutOfRangeException)(Function() Classify(0))
    End Sub
    <Fact>
    Public Sub NegativeIntegerIsRejectedNotANaturalNumber()
        Assert.Throws(Of ArgumentOutOfRangeException)(Function() Classify(-1))
    End Sub
End Class